### PR TITLE
Feature/exclude own checkin from matching

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
This PR adds a new field to Checkin (checkinSubmitted).
It is set to true after successful submission.
While matching, the users submitted checkins are ignored to prevent showing a risk based on the users own checkins.
Also some renamings and cleanup was added.

See tech spec change: https://github.com/corona-warn-app/cwa-app-tech-spec/pull/47/files

JIRA:
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6321

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Checkin.swift [hunk 1]
```diff
@@ -25,35 +25,22 @@ struct Checkin: Equatable {
 	let checkinEndDate: Date
 	let checkinCompleted: Bool
 	let createJournalEntry: Bool
-
-	var overlapInSeconds: Int = 0
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Checkin.swift:29]: is that not needed anymore?

File: src/xcode/ENA/ENA/Source/Scenes/Events/Store/__tests__/EventCheckoutServiceTests.swift [hunk 2]
```diff
@@ -360,37 +377,6 @@ class EventCheckoutServiceTests: XCTestCase {
 		XCTAssertEqual(mockUserNotificationCenter.notificationRequests.count, 1)
 	}
 
-	func makeDummyCheckin(
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Scenes/Events/Store/__tests__/EventCheckoutServiceTests.swift:363]: 👌

File: src/xcode/ENA/ENA/Source/Scenes/Events/Store/__tests__/CheckinSplittingServiceTests.swift [hunk 3]
```diff
@@ -177,33 +177,6 @@ class CheckinSplittingServiceTests: XCTestCase {
 		XCTAssertEqual("2021-03-05T08:15:00Z", utcFormatter.string(from: splitedCheckinEndDate1))
 	}
 
-	func makeDummyCheckin(
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Scenes/Events/Store/__tests__/CheckinSplittingServiceTests.swift:180]: 👌

File: src/xcode/ENA/ENA/Source/Scenes/Events/Store/__tests__/EventStoreTests.swift [hunk 4]
```diff
@@ -910,27 +917,6 @@ class EventStoreTests: XCTestCase {
 		)
 	}
 
-	private func makeCheckin(id: Int, checkinEndDate: Date = Date()) -> Checkin {
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Scenes/Events/Store/__tests__/EventStoreTests.swift:913]: 👌

File: src/xcode/ENA/ENA/Source/Scenes/Events/Store/__tests__/Model/CheckinTests.swift [hunk 5]
```diff
@@ -79,46 +79,14 @@ class CheckinTests: XCTestCase {
...
-		let checkin = makeDummyCheckin(
+		let checkin = Checkin.mock(
 			checkinStartDate: checkinStartDate,
 			checkinEndDate: checkinEndDate
 		)
 
 		XCTAssertEqual(checkin.roundedDurationIn15mSteps, 30)
 	}
 
-	func makeDummyCheckin(
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Scenes/Events/Store/__tests__/Model/CheckinTests.swift:90]: 👌

File: src/xcode/ENA/ENA/Source/Services/Event Registration/__tests__/CheckinRiskCalculationTests.swift [hunk 6]
```diff
@@ -183,18 +183,16 @@ class CheckinRiskCalculationTests: XCTestCase {
...
-				endDate: checkin1EndDate,
-				traceLocationId: "1"
+			Checkin.mock(
+				checkinStartDate: checkin1StartDate,
+				checkinEndDate: checkin1EndDate
 			)
 		)
 
 		let result2 = eventStore.createCheckin(
-			makeDummyCheckin(
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Services/Event Registration/__tests__/CheckinRiskCalculationTests.swift:194]: 👌

**Review by pascal-brause — COMMENTED** [review]
Nice that you killed about 6 times same creation of a dummy checkin. Great! 💎

**Review by pascal-brause — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Checkin.swift (hunk 1 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Checkin.swift:29]: Its not used anywhere.

**Review by marcussc — APPROVED** [review]

**Pull request merged**